operator/controllers/instance_controller.go

```diff
@@ -184,6 +192,36 @@ func (r *InstanceReconciler) valuesSecret(
 	return false, nil
 }
 
+// mergeValues merges the user-provided values and the default and static values provided by the
+// plan. The static values take precedence over the user-provided values, which in turn takes
+// precedence over the default values.
+func mergeValues(
+	instance *servicebrokerv1alpha1.Instance,
+	plan *servicebrokerv1alpha1.Plan,
+) ([]byte, error) {
+	userValues := make(map[string]interface{})
+	if err := yaml.Unmarshal([]byte(instance.Spec.Values), &userValues); err != nil {
+		return nil, err
```

Comment:
That `err` is going to be extremely unhelpful: which of the 3 values YAMLs failed to unmarshal?